pwr.md

```diff
@@ -0,0 +1,144 @@
+
+### core
+* `core_id`                 # core id number
+* `cpu`                     # this cores cpu object
+* `thread_siblings`         # list of thread siblings
+* `high_prioirty`           # high/normal priority
+* `base_freq`               # base frequency [2300Mhz]
+* `sst_bf_base_freq`        # priority based frequency [2100Mhz/2700Mhz]
+* `all_core_turbo_freq`     # all core turbo frequency [2800Mhz]
+* `highest_freq`            # highest frequency available [3900Mhz]
+* `lowest_freq`             # lowest active frequency [800Mhz]
+* `curr_freq`               # current core frequency
+* `min_freq`                # desired low frequency
+* `max_freq`                # desired high frequency
+* `epp`                     # energy performance preference
+
+The **core** object attributes which can be adjusted are `max_freq`, `min_freq` and `epp`, min and max set the desired core frequency at which an application wants the core to operate within. 
+>Available frequencies are between `lowest_freq` and `highest_freq`, with some defined freqeuncies being `base_freq`/`sst_bf_base_freq`/`all_core_turbo_freq` if configuration allows.
+
+`epp` sets the desired performance profile for the core.
+>Available profiles are `default`, `performance`, `balance_performance`, `balance_power` and `power`.
+
+The **cpu** object attributes which can be adjusted by the application are `uncore_max_freq` and `uncore_min_freq`, these set the desired uncore frequency at which an application wants the uncore to operate within.
+>Available frequencies are between `uncore_hw_max` and `uncore_hw_min`.
+
+```python
+core.min_freq = core.lowest_freq
+core.max_freq = core.highest_freq
+
+cpu.uncore_max_freq = cpu.uncore_hw_max
+cpu.uncore_min_freq = cpu.uncore_hw_min
+```
+
+### Committing 
+Modification of the power settings of a system is done by altering the core or CPU characteristics, as shown above, 
+then finalizing with the `commit()` function call, this will commit your changes and write to relevant MSRs or sysfs entries.
+```python 
+cores = pwr.get_cores()
+
+for core in cores:
+    core.min_freq = core.base_freq
+    core.max_freq = core.highest_freq
+    core.commit()
+```
+
+
+### Pre-set Profiles
+When an application is modifying the desired min and max core frequencies, pre-set configurations can also be applied, these will overwrite current configurations and commit the pre-sets.  
+* `minimum`:            Set all cores minimum to 800Mhz and maximum to 800Mhz.
+* `maximum`:            Set all cores minimum to 3900Mhz and maximum to 3900Mhz.
+* `base`:               Set all cores minimum to 2300Mhz and maximum to 2300Mhz.
+* `default`:            Set all cores minimum to 800Mhz and maximum to 3900Mhz.
+* `no_turbo`:           Set all cores minimum to 800Mhz and maximum to 2300Mhz.
+* `sst_bf_base`:        Set high priority cores minimum and maximum to 2700Mhz, normal priority cores minimum and maximum to 2100Mhz.
```

Comment:
I would add a note here that these profiles are valid only when SST-BF is enabled. They should probably be in a separate section as "additional profiles only available when SST-BF is enabled".